samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java

```diff
@@ -2938,19 +2938,17 @@ protected List<String> extractInstructionSegments(String instructions) {
 
   protected List<String> extractFeedbackSegments(String feedback) {
       List<String> segments = new ArrayList<>();
-      String[] results = null;
-
-      if (!StringUtils.isEmpty(feedback)) {
-          String[] firstResults = CALCQ_DOUBLE_OPEN_BRACKET.split(feedback); // only works because all variables and calculations are already replaced
-          for (String r:firstResults) {
-              results = (CALCQ_DOUBLE_CLOSE_BRACKET.split(r));
-              segments.addAll(Arrays.asList(results));
-          }
-          if (segments.size() == 1) {
-              // add in the trailing segment
-              segments.add("");
-          }
+      Pattern pattern = Pattern.compile(OPEN_BRACKET + OPEN_BRACKET + ".*?" + CLOSE_BRACKET + CLOSE_BRACKET);
```

Comment:
I'd move this into an init or something. There doesn't seem any point compiling this pattern every time this method is called.

Reply:
You are right! I'll move it. Thanks!

Reply:
Solved!